go-controller/pkg/ovn/pods.go

```diff
@@ -28,10 +28,14 @@ func (oc *Controller) syncPods(pods []interface{}) {
 			klog.Errorf("Spurious object in syncPods: %v", podInterface)
 			continue
 		}
-		_, err := util.UnmarshalPodAnnotation(pod.Annotations)
+		annotations, err := util.UnmarshalPodAnnotation(pod.Annotations)
```

Comment:
There is another case you have to handle here.

If a pod is created, an IP is allocated (either by ovn-kube or northd prior to upgrade), and then ovn-kube crashes, it will have an IP but no corresponding annotations.

In that case, you have to determine the IP from ovn and mark the IP as allocated to that pod, but not set the annotations, as the pod may not have been completely set up yet.

Reply:
Good point. Will add handling for it.

Reply:
done.